core/commands/remotepin.go

```diff
@@ -0,0 +1,203 @@
+var remotePinKey = os.Getenv("IPFS_REMOTE_PIN_KEY")
+
+var remotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Pin (and unpin) objects to remote pinning service.",
+	},
+
+	Subcommands: map[string]*cmds.Command{
+		"add": addRemotePinCmd,
+		"ls":  listRemotePinCmd,
+		"rm":  rmRemotePinCmd,
+	},
+}
+
+const pinNameOptionName = "name"
+const pinCIDsOptionName = "cid"
+
+type AddRemotePinOutput struct {
+	ID        string
+	Name      string
+	Delegates []multiaddr.Multiaddr
+}
+
+var addRemotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline:          "Pin objects to remote storage.",
+		ShortDescription: "Stores an IPFS object(s) from a given path to a remote pinning service.",
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("ipfs-path", true, false, "Path to object(s) to be pinned.").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption(pinNameOptionName, "An optional name for the pin."),
+	},
+	Type: AddRemotePinOutput{},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		opts := []pinclient.AddOption{}
+		if name, nameFound := req.Options[pinNameOptionName].(string); nameFound {
+			opts = append(opts, pinclient.PinOpts.WithName(name))
+		}
+
+		api, err := cmdenv.GetApi(env, req)
+		if err != nil {
+			return err
+		}
+
+		if len(req.Arguments) != 1 {
+			return fmt.Errorf("expecting one CID argument")
+		}
+		rp, err := api.ResolvePath(ctx, path.New(req.Arguments[0]))
+		if err != nil {
+			return err
+		}
+
+		c := pinclient.NewClient(remotePinURL, remotePinKey)
+
+		ps, err := c.Add(ctx, rp.Cid(), opts...)
+		if err != nil {
+			return err
+		}
+
+		for _, d := range ps.GetDelegates() {
+			p, err := peer.AddrInfoFromP2pAddr(d)
+			if err != nil {
+				return err
+			}
+			if err := api.Swarm().Connect(ctx, *p); err != nil {
+				log.Infof("error connecting to remote pin delegate %v : %w", d, err)
+			}
+
+		}
+
+		return res.Emit(&AddRemotePinOutput{
+			ID:        ps.GetId(),
+			Name:      ps.GetPin().GetName(),
+			Delegates: ps.GetDelegates(),
+		})
+	},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *AddRemotePinOutput) error {
+			fmt.Printf("pin_id=%v\n", out.ID)
+			fmt.Printf("pin_name=%q\n", out.Name)
+			for _, d := range out.Delegates {
+				fmt.Printf("pin_delegate=%v\n", d.String())
+			}
+			return nil
+		}),
+	},
+}
+
+var listRemotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "List objects pinned to remote pinning service.",
+		ShortDescription: `
+Returns a list of objects that are pinned to a remote pinning service.
+`,
+		LongDescription: `
+Returns a list of objects that are pinned to a remote pinning service.
+`,
+	},
+
+	Arguments: []cmds.Argument{},
+	Options: []cmds.Option{
+		cmds.StringOption(pinNameOptionName, "An optional name for the pin to be listed."),
```

Comment:
Reusing docs from https://ipfs.github.io/pinning-services-api-spec/#tag/pins/paths/~1pins/get:
```suggestion
		cmds.StringOption(pinNameOptionName, "Return pins objects with names that contain provided value (case-insensitive, partial or full match)."),
```